python/ray/train/_internal/logging.py

```diff
@@ -0,0 +1,239 @@
+import logging
+import logging.config
+import os
+from enum import Enum
+from typing import Optional
+
+import yaml
+
+import ray
+from ray.train._internal.session import _get_session
+
+# JSON Encoding format for Ray Train structured logging
+DEFAULT_JSON_LOG_ENCODING_FORMAT = "JSON"
+# Env. variable to specify the encoding of the file logs when using the default config.
+LOG_ENCODING_ENV = "RAY_TRAIN_LOG_ENCODING"
```

Comment:
`RAY_TRAIN_LOG_ENCODING` now controls the **console output encoding format** between TEXT and JSON, but I think it should always be `TEXT`.

Our product should probably never set console output to JSON mode automatically, and users should not know about this environment variable.

What about this:
* Remove this `RAY_TRAIN_LOG_ENCODING` environment variable, so that console is always TEXT, and file (`ray-train.log`) is always JSON.

Reply:
I see the point cc @justinvyu . There is actually a longer story why I changed to allow JSON mode in console after discussed with @matthewdeng .

1. I think the general logging is moving towards deprecate `log_to_driver` feature, that the logs of all works will not be showing in the driver console, but persisted in their local files, i.e., *.err, *.std, *.log. These logs will be ingested and showed using more modern tools like log viewer eventually. In the long run, logging to driver console will not be that useful.
2. The ray core logger has only one stream handler that writes everything to the console, i.e., *.err file of each node by default. These *.err console file are expected to contain JSON logs anyway if ray core structured logging is enabled. We are also enabling JSON mode of console logs following ray core's pattern.

There could be another design pattern as you suggested. (1) Everything to the console must be TEXT; (2) All the JSON logs go to a separate file `ray-core.log`, `ray-train.log` ...etc, if JSON mode is enabled. (3) console log files, *.err, *.std will never be ingested because they are never JSON. I think this also works but may need wider revamp of many ray libraries. We can have a chat on this tomorrow.